### PR TITLE
Bug 1853687 - run Rust work on test thread so it can complete sequentially

### PR DESCRIPTION
**Pull request by MatthewTighe**
I think the underlying issue with the original bug was that we are switching coroutine context to be on a separate thread in `PlacesStorage`. Hopefully this will mean that all the work can complete sequentially and the tests will behave correctly again.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1853687

### CODE REVIEW

**Review by HarrisonOg — APPROVED** [review]
looks good to me!

File: android-components/components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/PlacesHistoryStorageTest.kt [hunk 1]
```diff
@@ -545,9 +544,9 @@ class PlacesHistoryStorageTest {
         assertEquals(0, visits.size)
     }
 
-    @Ignore("Disabled: https://bugzilla.mozilla.org/show_bug.cgi?id=1853687")
     @Test
     fun `store can delete by 'range'`() = runTestOnMain {
+        history.writeScope = this
```
Comment (jonalmeida) [inline android-components/components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/PlacesHistoryStorageTest.kt:549]: Drive-by 🚗 : Should we do this in the setup so it's not a footgun for future tests?

Comment (MatthewTighe) [reply android-components/components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/PlacesHistoryStorageTest.kt:549]: I can file a follow up, but I wasn't sure without a deeper investigation whether the test scope for setup would carry over to the test function itself

Comment (MatthewTighe) [reply android-components/components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/PlacesHistoryStorageTest.kt:549]: This should allow us to test whether this fixes the flaky behavior at least. If it does I will move forward with your suggestion, thanks!

Comment (MatthewTighe) [reply android-components/components/browser/storage-sync/src/test/java/mozilla/components/browser/storage/sync/PlacesHistoryStorageTest.kt:549]: https://bugzilla.mozilla.org/show_bug.cgi?id=1858240

**Pull request merged**